### PR TITLE
WinRM Connection Pooling implementation

### PR DESCRIPTION
**Pull request by narayanan**
We noticed that HTTPClient used for WinRM connections are not pooled. Updated the client to use a connection pool manager.

Tested the changes in our internal jenkins instances.

### CODE REVIEW

**Comment by narayanan** [comment]
Just noticed this is very similar to other PR which is in draft: https://github.com/jenkinsci/ec2-plugin/pull/469

File: src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java [hunk 1]
```diff
@@ -75,7 +66,7 @@
     private final ThreadLocal<BasicAuthCache> authCache = new ThreadLocal<BasicAuthCache>();
     private boolean useHTTPS;
     private BasicCredentialsProvider credsProvider;
-    private boolean allowSelfSignedCertificate;
+    private boolean     allowSelfSignedCertificate;
```
Comment (slide) [inline src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java:69]: Weird whitespace addition

Comment (narayanan) [reply src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java]: Fixed the spacing

**Review by res0nance — APPROVED** [review]

**Comment by res0nance** [comment]
@narayanan Could you rebase this?

**Comment by narayanan** [comment]
@res0nance Resolved conflicts

**Pull request merged**